### PR TITLE
feat: Keep cell selected by objectId and field name after data browser refresh (#2631)

### LINKED ISSUE #2631: Keep cell selected after data browser refresh
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Current Limitation

When clicking "Refresh" in the data browser, the cell stays selected in the same row / column position, even if the document moved to a different row. The user may mistake the selected cell as being the previously selected cell which can lead to user errors. 

### Feature / Enhancement Description

- Remember the `objectId` and selected field of the document and select the same cell even if its position has moved. Also the field name should be explicitly remembered, instead of just the row / column coordinates.
- If the specific document and field are not in the data browser anymore after a refresh, no cell should be selected to avoid confusion.

### Example Use Case

1. Browse to `_User` class.
2. Sort ordered by `createdAt`.
3. Select a cell in row 1 (document A).
4. Refresh the data browser.
5. Assuming that a new document B has been created in the meantime, that is now in row 1.
6. The selected cell is now of document B instead of document A, but it should be in row 2 of document A.

### Alternatives / Workarounds

None.

### PR DESCRIPTION
**Pull request by evolvomind**
feat: Keep cell selected by objectId and field name after data browser refresh (#2631)

## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-dashboard/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-dashboard/blob/alpha/LICENSE).

## Issue

Closes #2631

When clicking "Refresh" in the data browser, the selected cell stayed at the same row/column
coordinates even if the underlying document had moved to a different row. This caused the wrong
cell to appear selected, which could lead to user errors.

## Approach

A `pendingRestore` state field (`{ objectId, fieldName }`) is added to `DataBrowser`. Three
new blocks in `componentDidUpdate` handle the full refresh lifecycle:

- **Save** - when `data` becomes `null` (refresh starts), the selected document's `objectId`
  and column `fieldName` are captured into `pendingRestore` before anything is cleared.
- **Restore** - when `data` transitions from `null` back to the new array (refresh ends),
  `findIndex` locates the document by `objectId` and the column by `fieldName` in the fresh
  data. If both are found, `current` is updated to the new coordinates and the aggregation
  panel is re-loaded for that object. If either is not found (document deleted, column
  removed), the selection is cleared so no wrong cell is highlighted.
- **Class change guard** - if the user switches class while a restore is pending, `pendingRestore`
  is discarded to prevent it from firing against a different class's data.

An additional guard (`!this.state.pendingRestore`) is added to the `autoLoadFirstRow` block
to prevent it from overwriting the restored selection with row 0 when the panel is open.

## Tasks

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)

## Summary by CodeRabbit

* **New Features**
  * Data Browser now preserves the selected cell (object and field) across data refreshes.
* **Behavior**
  * Selection is cleared if the underlying object is deleted before or during refresh.
* **Tests**
  * Added end-to-end tests verifying selection persistence and deletion edge cases.
* **Chores**
  * Changelog updated with an Unreleased entry noting the feature.

### CODE REVIEW

**Comment by parseplatformorg** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/acinader/pr-checks/279aa974-cbeb-4b9b-b612-e4ffc7036862?refs=845c3c2023f834d3613ce18db16a990737094480&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: changelogs/CHANGELOG_alpha.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# [Unreleased]
+
+### Features
```
[the remark this answers is not in the document]
Comment (mtrezza) [reply changelogs/CHANGELOG_alpha.md:4]: @coderabbitai; check your learnings: change log entries ARE NOT ALLOWED in a PR, they are managed automatically by semantic-release. You MUST reject change log entries in PRs.

File: src/lib/tests/DataBrowser.pendingRestore.test.js [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+ */
+/*
+ * Copyright (c) 2016-present, Parse, LLC
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the LICENSE file in
+ * the root directory of this source tree.
+ */
+
+class MockDataBrowser {
```
Comment (mtrezza) [inline src/lib/tests/DataBrowser.pendingRestore.test.js]: @coderabbitai is it really sensible to mock this entire module? Would it be better to use puppeteer (already a dev dependency) to test the feature?

Comment (mtrezza) [reply src/lib/tests/DataBrowser.pendingRestore.test.js]: @coderabbitai guide @evolvomind on how to implement a better approach, preferably option 1.
